# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/i386.patch
# Copyright (C) 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

syscalls.s/ppoll.S:3: Warning: 309 shortened to 53
syscalls.s/openat.S:3: Warning: 295 shortened to 39

--- dietlibc-0.32/i386/syscalls.h.vanilla	2011-01-09 16:42:52.627095848 +0000
+++ dietlibc-0.32/i386/syscalls.h	2011-01-09 16:43:40.067486265 +0000
@@ -335,8 +335,13 @@
 .type sym,@function; \
 .global sym; \
 sym: \
+.ifle __NR_##name-255; \
 	movb $__NR_##name,%al; \
 	jmp __unified_syscall; \
+.else; \
+	movw $__NR_##name,%ax; \
+	jmp __unified_syscall_256; \
+.endif; \
 .Lend##sym: ; \
 .size sym,.Lend##sym-sym
 
